Build failure with GHC 9.0.1

```
src/Foreign/Lua/Core/Error.hs:80:8: error:
    • Couldn't match type: forall a1. Lua.State -> IO a1
                     with: Lua.State -> IO a
      Expected: Lua.ErrorConversion -> Lua.State -> IO a
        Actual: Lua.ErrorConversion -> forall a. Lua.State -> IO a
    • In the first argument of ‘(<$>)’, namely ‘Lua.errorToException’
      In a stmt of a 'do' block:
        f <- Lua.errorToException <$> Lua.errorConversion
      In the expression:
        do f <- Lua.errorToException <$> Lua.errorConversion
           l <- Lua.state
           Lua.liftIO (f l)
    • Relevant bindings include
        throwErrorAsException :: Lua a
          (bound at src/Foreign/Lua/Core/Error.hs:79:1)
   |
80 |   f <- Lua.errorToException <$> Lua.errorConversion
   |        ^^^^^^^^^^^^^^^^^^^^

src/Foreign/Lua/Core/Error.hs:96:41: error:
    • Couldn't match type: forall a1. Lua.State -> IO a1
                     with: Lua.State -> IO a
      Expected: Lua.ErrorConversion -> Lua.State -> IO a
        Actual: Lua.ErrorConversion -> forall a. Lua.State -> IO a
    • In the second argument of ‘(.)’, namely ‘Lua.errorToException’
      In the second argument of ‘(>>=)’, namely
        ‘Lua.liftLua . Lua.errorToException’
      In a stmt of a 'do' block:
        Lua.errorConversion >>= Lua.liftLua . Lua.errorToException
    • Relevant bindings include
        throwMessage :: String -> Lua a
          (bound at src/Foreign/Lua/Core/Error.hs:92:1)
   |
96 |   Lua.errorConversion >>= Lua.liftLua . Lua.errorToException
   |                                         ^^^^^^^^^^^^^^^^^^^^

src/Foreign/Lua/Core/Error.hs:101:12: error:
    • Couldn't match type: forall a1. Lua a1 -> Lua a1 -> Lua a1
                     with: Lua a -> Lua a -> Lua a
      Expected: Lua.ErrorConversion -> Lua a -> Lua a -> Lua a
        Actual: Lua.ErrorConversion -> forall a. Lua a -> Lua a -> Lua a
    • In the first argument of ‘(<$>)’, namely ‘Lua.alternative’
      In a stmt of a 'do' block:
        alt <- Lua.alternative <$> Lua.errorConversion
      In the expression:
        do alt <- Lua.alternative <$> Lua.errorConversion
           x `alt` y
    • Relevant bindings include
        y :: Lua a (bound at src/Foreign/Lua/Core/Error.hs:100:9)
        x :: Lua a (bound at src/Foreign/Lua/Core/Error.hs:100:3)
        (<|>) :: Lua a -> Lua a -> Lua a
          (bound at src/Foreign/Lua/Core/Error.hs:100:5)
    |
101 |     alt <- Lua.alternative <$> Lua.errorConversion
    |            ^^^^^^^^^^^^^^^
```

--- a/src/Foreign/Lua/Core/Auxiliary.hs
+++ b/src/Foreign/Lua/Core/Auxiliary.hs
@@ -227,11 +227,11 @@ ref t = liftLua $ \l -> Lua.toReference <$> luaL_ref l t
 tostring' :: StackIndex -> Lua B.ByteString
 tostring' n = do
   l <- Lua.state
-  e2e <- Lua.errorToException <$> Lua.errorConversion
+  e <- Lua.errorConversion
   Lua.liftIO $ alloca $ \lenPtr -> do
     cstr <- hsluaL_tolstring l n lenPtr
     if cstr == nullPtr
-      then e2e l
+      then Lua.errorToException e l
       else do
         cstrLen <- Storable.peek lenPtr
         B.packCStringLen (cstr, fromIntegral cstrLen)

--- a/src/Foreign/Lua/Core/Error.hs
+++ b/src/Foreign/Lua/Core/Error.hs
@@ -77,9 +77,9 @@ try = Catch.try
 -- expected to be at the top of the stack.
 throwErrorAsException :: Lua a
 throwErrorAsException = do
-  f <- Lua.errorToException <$> Lua.errorConversion
+  e <- Lua.errorConversion
   l <- Lua.state
-  Lua.liftIO (f l)
+  Lua.liftIO (Lua.errorToException e l)
 
 -- | Alias for `throwErrorAsException`; will be deprecated in the next
 -- mayor release.
@@ -93,13 +93,14 @@ throwMessage msg = do
   Lua.liftLua $ \l ->
     B.unsafeUseAsCStringLen (Utf8.fromString msg) $ \(msgPtr, z) ->
       lua_pushlstring l msgPtr (fromIntegral z)
-  Lua.errorConversion >>= Lua.liftLua . Lua.errorToException
+  e <- Lua.errorConversion
+  Lua.liftLua (Lua.errorToException e)
 
 instance Alternative Lua where
   empty = throwMessage "empty"
   x <|> y = do
-    alt <- Lua.alternative <$> Lua.errorConversion
-    x `alt` y
+    e <- Lua.errorConversion
+    Lua.alternative e x y
 
 -- | Convert the object at the top of the stack into a string and throw
 -- it as a HsLua @'Exception'@.

